packages/renderer/src/lib/ui/TreeView.svelte

```diff
@@ -0,0 +1,130 @@
+<style>
+.no-arrow {
+  padding-left: 1rem;
+}
+.arrow {
+  cursor: pointer;
+  display: inline-block;
+}
+.arrowDown {
+  transform: rotate(90deg);
+}
+</style>
+
+<script context="module" lang="ts">
+const _expansionState = new Map<string, boolean>();
+</script>
+
+<script lang="ts">
+import type { fileNode } from '../../../../main/src/plugin/filetree';
+
+export let tree: fileNode<any>;
+export let margin = 0;
+export let root = true;
+
+$: label = isRemoved(tree) ? tree.name.substring(4) : tree.name;
+$: children = tree.children;
+$: file = tree.data;
+$: colorClass = getColor(tree.name, tree.data);
+
+function getColor(name: string, data: any) {
+  if (!data) {
+    return '';
+  }
+  if (name.startsWith('.wh.')) {
+    return 'text-red-500';
+  }
+  if (data.type === 'SymbolicLink') {
+    return 'text-sky-300';
+  }
+  if (data.type === 'Directory') {
+    return 'text-sky-500';
+  }
+  if (isExecutable(data.mode)) {
+    return 'text-green-500';
+  }
```

Comment:
it does not look like a generic TreeView component

AFAIK the component should not be tied to 'SymbolicLink' or 'Directory' or '.wh' or some executables

color should be provided by the consumer of the component, not the component
component should take leafs/tree as input